src/main.rs

```diff
@@ -50,6 +53,9 @@ impl<'a> Mqtt<'a> {
             keep_alive: Duration::from_secs(5),
             cap: 10,
             clean_session: false,
+            availability_topic: "gpio/availability",
```

Comment:
It's a bit unfortunate that this is never used, as it will be overriden by either the setting in `config.yaml` or the default from `config.rs` - simliar to the other settings here.

I think this shoudl at least reflect the same defaults, so I suggest setting this to the same value as the default:

```suggestion
            availability_topic: "rpi-mqtt-gpio/status",
```

(I'm not quite satisfied with the overall approach - something using `Option<>` for non-obligatory settings might be nicer, overriding them with their defaults in `config.rs` shouldn't they be set. But this is a topic for a seperate pull request one day 👀 )